### PR TITLE
Add a command/action to clear history

### PR DESCRIPTION
**Pull request by guywald1**
This PR adds a command and a button/action to clear the reference history.

Closes https://github.com/microsoft/vscode/issues/73548.

Couldn't find a relevant icon so I made a custom one (top right, second one from the right):

image

> I don't know how contributers normally approach icons, so any feedback would be great.

### CODE REVIEW

File: src/extension.ts [hunk 1]
```diff
@@ -144,6 +144,11 @@ export function activate(context: vscode.ExtensionContext) {
         });
     }
 
+    const clearHistoryCommand = async () => {
+        await clearCommand();
+        history.clear();
```
Comment (jrieken) [inline src/extension.ts]: after clearing the history, the message still reads _"To populate this view, open an editor and run the 'Find All References'-command or run a previous search again:"_ which isn't correct anymore. Instead it should show the default message

File: media/action-clear-history.svg [hunk 2]
```diff
@@ -0,0 +1 @@
...10.4,6c0.6,0.6,0.9,1.3,0.9,2.1c0,0.8-0.3,1.5-0.9,2.1		c-1.2,1.2-3.1,1.2-4.2,0L6,9.9L5.9,9.7L5.8,9.6C4.6,8.7,3.5,8.5,3,8.4l-0.1,0L1.5,8.6l0.3,1.1C2,10.9,2.7,12,3.5,12.9z"/>		<path fill="#A1260D" d="M9.1,11.4c-0.6,0-1.2-0.2-1.6-0.7L6.5,9.7C6.2,9.5,6.1,9.2,6,8.9C5.9,8.7,5.8,8.4,5.8,8.1V5.9			c0-1.2,1-2.2,2.2-2.2c1.2,0,2.2,1,2.2,2.2v1.2l0.4,0.4c0.4,0.4,0.7,1,0.7,1.6s-0.2,1.2-0.7,1.6S9.7,11.4,9.1,11.4z"/>		<path fill="#F6F6F6" d="M8.1,4.7c0.7,0,1.2,0.6,1.2,1.2v1.6L10,8.2c0.5,0.5,0.5,1.3,0,1.8c-0.2,0.2-0.5,0.4-0.9,0.4			c-0.3,0-0.6-0.1-0.9-0.4L7.2,9C7.1,8.8,7,8.7,6.9,8.6C6.8,8.4,6.8,8.2,6.8,8.1V5.9C6.8,5.2,7.4,4.7,8.1,4.7 M8.1,2.7			c-1.8,0-3.2,1.5-3.2,3.2v2.1c0,0.4,0.1,0.8,0.2,1.2c0.2,0.4,0.4,0.8,0.7,1.1l1.1,1.1c0.6,0.6,1.4,1,2.3,1s1.7-0.3,2.3-1			c0.6-0.6,1-1.4,1-2.3c0-0.9-0.3-1.7-1-2.3l-0.1-0.1V5.9C11.3,4.1,9.8,2.7,8.1,2.7L8.1,2.7z"/>	<path fill="#424242" d="M4.3,12.2c2.2,2.2,5.9,2.2,8.1,0c1.1-1.1,1.7-2.6,1.7-4.1c0-1.5-0.6-3-1.7-4.1c-2.2-2.2-5.9-2.2-8.1,0		L4,4.3L2.4,2.7L2,7.2L4.5,7c0,0,2.1-0.2,2.1-0.2L5.3,5.5l0.2-0.3c1.6-1.6,4.1-1.6,5.7,0c0.8,0.8,1.2,1.8,1.2,2.8c0,1-0.4,2-1.2,2.8		c-1.6,1.6-4.1,1.6-5.7,0c0,0-0.2-0.2-0.3-0.5c-1-0.8-1.9-1-2.3-1l-0.1,0C3,10.4,3.5,11.4,4.3,12.2z"/></svg>
```
Comment (jrieken) [inline media/action-clear-history.svg]: use the same clear icon that's used for clearing results

File: package.json [hunk 3]
```diff
@@ -136,6 +145,11 @@
           "command": "references-view.clear",
           "group": "navigation",
           "when": "view == references-view.tree && reference-list.hasResult"
+        },
+        {
+          "command": "references-view.clearHistory",
+          "group": "navigation",
+          "when": "!reference-list.hasResult"
```
Comment (jrieken) [inline package.json]: maybe have a context key for showing clear history only when having a history

File: media/action-clear-history.svg (hunk 2 above)
Comment (guywald1) [reply media/action-clear-history.svg]: Wanted to do that, but it seemed like a strange UX that you have an icon to clear results and after clicking it, you have the _same_ icon to clear history. No?

**Comment by jrieken** [comment]
Screenshot 2019-05-13 at 09 55 05

Seeing two clear icons now...

**Comment by guywald1** [comment]
@jrieken Yeah, desired behavior but it might be better to only show it when a result isn't selected. Especially if I change the icon to be the same one (WDYT about that btw? Haven't gotten a reply in the thread)

**Comment by guywald1** [comment]
@jrieken For now I changed the icon back and changed it so it doesn't show when there's a result selected.
Though I personally think a different icon is better UX,  but I may be wrong here.

**Comment by jrieken** [comment]
> Though I personally think a different icon is better UX, but I may be wrong here.

I understand but we try to go for persistency and use the same icon for the same/similar operation

**Comment by jrieken** [comment]
changes look good and work well. thanks!

**Comment by m-wheeler** [comment]
Looking forward to it in the next visual studio update!

**Comment by jrieken** [comment]
Use Insiders and have it today: https://code.visualstudio.com/insiders/

**Pull request merged**